examples/GLUE/src/data.py

```diff
@@ -0,0 +1,128 @@
+from pymarlin.core.data_interface import DataInterface, DataProcessor
+from datasets import load_dataset
+import matplotlib.pyplot as plt
+import pandas as pd
+
+cache_dir  =r"/tmp/hf_data"
+class GlueDataAnalyzer(DataProcessor):
+    def __init__(self, glue_task):
+        self.datasets = load_dataset("glue",glue_task, cache_dir = cache_dir)
+    
+    def process(self):
```

Comment:
I see - these processors aren't actually processing and are just used to analyze the splits. This is our first case of using HF datasets and I think this could even prompt a discussion of the purpose of DataProcessor / DataInterface.

Reply:
Not all datasets come  from huggingface. There will be numerous cases where people will want to train a model on a custom dataset like college projects, enterprise data etc. We should give users the freedom to chose whatever they want.

Reply:
yes we definitely want the ability to still process and read from file, I think we should definitely keep data processor and data interface for dealing with those and we just merged ashwin's change to process data in processor instead of intercace.